sdk/src/wallet/operations/transaction/high_level/delegation/create.rs

```diff
@@ -0,0 +1,149 @@
+#[serde(rename_all = "camelCase")]
+pub struct CreateDelegationTransaction {
+    pub delegation_id: DelegationId,
+    pub transaction: TransactionWithMetadata,
+}
+
+/// The result of preparing a transaction to create a delegation
+#[derive(Debug, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PreparedCreateDelegationTransaction {
+    pub delegation_id: DelegationId,
+    pub transaction: PreparedTransactionData,
+}
+
+impl<S: 'static + SecretManage> Wallet<S>
+where
+    crate::wallet::Error: From<S::Error>,
+    crate::client::Error: From<S::Error>,
+{
+    /// Creates a delegation output.
+    /// ```ignore
+    /// let params = CreateDelegationParams {
+    ///     address: None,
+    ///     delegated_amount: 200_000,
+    ///     validator_address: AccountAddress::from_str("0xe1d4bad757d5180811ab81f6c014bb2d66c152efe56cf7a467047625b0016868")?,
+    /// };
+    ///
+    /// let tx: CreateDelegationTransaction = delegation.create_delegation_output(params, None).await?;
+    /// println!(
+    ///     "Transaction sent: {}/transaction/{}",
+    ///     std::env::var("EXPLORER_URL").unwrap(),
+    ///     tx.transaction.transaction_id
+    /// );
+    /// ```
+    pub async fn create_delegation_output(
+        &self,
+        params: CreateDelegationParams,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<CreateDelegationTransaction> {
+        let options = options.into();
+        let prepared = self.prepare_create_delegation_output(params, options.clone()).await?;
+
+        self.sign_and_submit_transaction(prepared.transaction, None, options)
+            .await
+            .map(|transaction| CreateDelegationTransaction {
+                delegation_id: prepared.delegation_id,
+                transaction,
+            })
+    }
+
+    /// Prepares the transaction for [Wallet::create_delegation_output()].
+    pub async fn prepare_create_delegation_output(
+        &self,
+        params: CreateDelegationParams,
+        options: impl Into<Option<TransactionOptions>> + Send,
+    ) -> crate::wallet::Result<PreparedCreateDelegationTransaction> {
+        log::debug!("[TRANSACTION] prepare_create_delegation_output");
+
+        let address = match params.address.as_ref() {
+            Some(bech32_address) => {
+                self.client().bech32_hrp_matches(bech32_address.hrp()).await?;
+                bech32_address.inner().clone()
+            }
+            None => self.address().await.inner().clone(),
+        };
+
+        let output = DelegationOutputBuilder::new_with_amount(
+            params.delegated_amount,
+            DelegationId::null(),
+            params.validator_address,
+        )
+        .add_unlock_condition(AddressUnlockCondition::new(address.clone()))
+        .finish_output()?;
+
+        let transaction = self.prepare_transaction([output], options).await?;
+
+        Ok(PreparedCreateDelegationTransaction {
```

Comment:
Why do we need to return the delegationID?

Reply:
We don't do it for the other chains

Reply:
just because it's annoying to get otherwise? I dunno